File: mailwaycore/src/main/java/com/dimension/mailwaycore/data/entity/ContactChannel.kt
package com.dimension.mailwaycore.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.Serializable

@Entity
@Serializable
data class ContactChannel(
    @PrimaryKey var id: String,
    var name: String,
    var value: String,
    var created_at: Long,
    var updated_at: Long,
    var contactId: String
)

File: mailway_ui/lib/data/entity/ContactChannel.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:mailwayui/data/entity/Contact.dart';

part 'ContactChannel.g.dart';

@JsonSerializable()
class ContactChannel {
  String id;
  String name;
  String value;
  int created_at;
  int updated_at;
  String contactId;

  ContactChannel({
    this.id,
    this.name,
    this.value,
    this.created_at,
    this.updated_at,
    this.contactId,
  });

  factory ContactChannel.fromJson(Map<String, dynamic> json) =>
      _$ContactChannelFromJson(json);

  Map<String, dynamic> toJson() => _$ContactChannelToJson(this);
}
